### PR TITLE
Misc

### PR DESCRIPTION
**Pull request by jschueller**
- #2274
- #2255 
- #2229
- #2205
- #2287
- #2285
- #2281
- add Basis::getInputDimension/getOutputDimension and deprecate confusing getDimension (returned dim of input)

### CODE REVIEW

File: lib/src/Base/Func/FiniteBasis.cxx [hunk 1]
```diff
@@ -43,7 +43,19 @@ FiniteBasis::FiniteBasis(const FunctionCollection & collection)
...
+    const UnsignedInteger inputDimension = collection[0].getInputDimension();
+    const UnsignedInteger outputDimension = collection[0].getOutputDimension();
+    for (UnsignedInteger i = 1; i < size; ++ i)
+    {
+      if (collection[i].getInputDimension() != inputDimension)
+        throw InvalidArgumentException(HERE) << "Basis elements cannot have different input dimensions";
+      if (collection[i].getOutputDimension() != outputDimension)
+        throw InvalidArgumentException(HERE) << "Basis elements cannot have different output dimensions";
+    }
+  }
```
Comment (sofianehaddad) [inline lib/src/Base/Func/FiniteBasis.cxx:58]: Good
Maybe we should have specific accessors ?
The `Basis::getDimension` is confusing

Comment (jschueller) [reply lib/src/Base/Func/FiniteBasis.cxx:58]: yes, lets add getInputDimension and getOutputDimension and deprecate getDimension

**Review by sofianehaddad — APPROVED** [review]

**Comment by jschueller** [comment]
@regislebrun this one is ready, waiting for your review

**Review by regislebrun — APPROVED** [review]
So many files changed! I hope some tasks have been automated ;-)

**Pull request merged**